packages/client-core/src/social/services/FriendService.ts

```diff
@@ -1,50 +1,56 @@
 import { Paginated } from '@feathersjs/feathers'
-import { createState, none, useState } from '@speigg/hookstate'
+import { none } from '@speigg/hookstate'
 import _ from 'lodash'
 
 import { User } from '@xrengine/common/src/interfaces/User'
 import { UserRelationship } from '@xrengine/common/src/interfaces/UserRelationship'
+import { matches, Validator } from '@xrengine/engine/src/common/functions/MatchesUtils'
+import { addActionReceptor, defineAction, defineState, dispatchAction, getState, useState } from '@xrengine/hyperflux'
 
 import { NotificationService } from '../../common/services/NotificationService'
 import { client } from '../../feathers'
-import { store, useDispatch } from '../../store'
+import { store } from '../../store'
 import { accessAuthState } from '../../user/services/AuthService'
 import { UserAction } from '../../user/services/UserService'
 
 //State
-const state = createState({
-  friends: {
-    friends: [] as Array<User>,
-    total: 0,
-    limit: 5,
-    skip: 0
-  },
-  getFriendsInProgress: false,
-  updateNeeded: true
+const FriendState = defineState({
+  name: 'FriendState',
+  initial: () => ({
+    friends: {
+      friends: [] as Array<User>,
+      total: 0,
+      limit: 5,
+      skip: 0
+    },
+    getFriendsInProgress: false,
+    updateNeeded: true
+  })
 })
 
-store.receptors.push((action: FriendActionType): any => {
-  let newValues, selfUser, otherUser, otherUserId
-  state.batch((s) => {
-    switch (action.type) {
-      case 'LOADED_FRIENDS':
+export const FriendServiceReceptor = (action) => {
+  getState(FriendState).batch((s) => {
+    let newValues, selfUser, otherUser, otherUserId
```

Comment:
let's take the opportunity to scope these to each `when` to ensure we have types for all of these